packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,509 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Method,
+  Prop,
+  State,
+  VNode,
+  Watch,
+} from "@stencil/core";
+import { focusFirstTabbable, toAriaBoolean } from "../../utils/dom";
+import {
+  activateFocusTrap,
+  connectFocusTrap,
+  deactivateFocusTrap,
+  FocusTrap,
+  FocusTrapComponent,
+  updateFocusTrapElements,
+} from "../../utils/focusTrapComponent";
+import {
+  componentFocusable,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent,
+} from "../../utils/loadable";
+import { createObserver } from "../../utils/observers";
+import { onToggleOpenCloseComponent, OpenCloseComponent } from "../../utils/openCloseComponent";
+import { Kind, Scale } from "../interfaces";
+import { connectLocalized, disconnectLocalized, LocalizedComponent } from "../../utils/locale";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages,
+} from "../../utils/t9n";
+import { componentOnReady } from "../../utils/component";
+import { SLOTS as PANEL_SLOTS } from "../panel/resources";
+import { HeadingLevel } from "../functional/Heading";
+import { OverlayPositioning } from "../../components";
+import { DialogMessages } from "./assets/dialog/t9n";
+import { CSS, SLOTS } from "./resources";
+import { DialogPlacement } from "./interfaces";
+
+let totalOpenDialogs: number = 0;
+let initialDocumentOverflowStyle: string = "";
+
+/**
+ * @slot - A slot for adding content.
+ * @slot content - A slot for adding custom content.
+ * @slot action-bar - A slot for adding a `calcite-action-bar` to the component.
+ * @slot alerts - A slot for adding `calcite-alert`s to the component.
+ * @slot content-bottom - A slot for adding content below the unnamed (default) slot and above the footer slot (if populated)
```

Comment:
nitpick: 

Add a period at the end of the `content-bottom` slot description and a minor copyedit and backticks for the slot:

> A slot for adding content below the unnamed (default) slot and - if populated - above the `footer` slot.